<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>view::HoleFillWindow</class>
 <widget class="QMainWindow" name="view::HoleFillWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>938</width>
    <height>650</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Hole Filling</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="toolTip">
       <string>Click on the image to mark a &quot;hole&quot; to be filled using PatchMatch.</string>
      </property>
      <property name="widgetResizable">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignmentFlag::AlignCenter</set>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>398</x>
         <y>168</y>
         <width>118</width>
         <height>78</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Master Controls</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_4">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Hole Brush Width:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="brushWidthSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>1000</number>
           </property>
           <property name="value">
            <number>20</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="clearHoleButton">
         <property name="text">
          <string>Clear Hole</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="startManualButton">
         <property name="toolTip">
          <string>Enter mode where you can manually step through applying PatchMatch to fill the marked pixels.</string>
         </property>
         <property name="text">
          <string>Manual Hole Fill (CPU)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="automaticFillButton">
         <property name="toolTip">
          <string>Use the CPU (non-OpenCL) implementation of PatchMatch to populate the marked pixels.</string>
         </property>
         <property name="text">
          <string>Automatic Fill (CPU)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="automaticFillOpenCLButton">
         <property name="toolTip">
          <string>Use the OpenCL implementation of PatchMatch to populate the marked pixels.</string>
         </property>
         <property name="text">
          <string>Automatic Fill (OpenCL)</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>529</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="manualFillGroupBox">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="title">
       <string>Manual Hole Fill</string>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QPushButton" name="refineNNFPushButton">
         <property name="toolTip">
          <string>Improve the NNF by performing a search and a propagation (repeat &quot;Steps Per Refine&quot; times).</string>
         </property>
         <property name="text">
          <string>Refine NNF</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>Search/Prop Steps Per Refine:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="numRefinesSpinBox">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="value">
            <number>1</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPushButton" name="nextPyramidButton">
         <property name="toolTip">
          <string>Go to the next pyramid level</string>
         </property>
         <property name="text">
          <string>Go to Pyramid Level:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="quitManualButton">
         <property name="text">
          <string>Quit Manual Fill</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Orientation::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>207</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>938</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionOpenCLDevice"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuSettings"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionLoad">
   <property name="text">
    <string>Load Image</string>
   </property>
  </action>
  <action name="actionOpenCLDevice">
   <property name="text">
    <string>OpenCL Device</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
